gio/src/socket_listener.rs

```diff
@@ -0,0 +1,55 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::{pin::Pin, task::ready};
+
+use futures_core::{
+    stream::Stream,
+    task::{Context, Poll},
+    Future,
+};
+
+use crate::{prelude::SocketListenerExt, SocketConnection, SocketListener};
+use glib::{prelude::*, Error, Object};
+
+pub struct Incoming {
+    listener: SocketListener,
+    fut: Option<Pin<Box<dyn Future<Output = Result<(SocketConnection, Option<Object>), Error>>>>>,
+}
+
+impl Stream for Incoming {
+    type Item = Result<(SocketConnection, Option<Object>), Error>;
+
+    fn poll_next(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        if self.fut.is_none() {
+            self.fut = Some(self.listener.accept_future());
+        }
+
+        let fut = self.fut.as_mut().unwrap();
+        let res = ready!(Pin::new(fut).poll(ctx));
+        self.fut.take();
+
+        Poll::Ready(Some(res))
+    }
+}
+
+pub trait SocketListenerExtManual: SocketListenerExt {
+    /// Returns a stream of incoming connections
```

Comment:
```suggestion
    // rustdoc-stripper-ignore-next
    /// Returns a stream of incoming connections
```

and then the CI will also be happy :)